src/ranking.jl

```diff
@@ -168,7 +171,23 @@ of an array. The `lt` keyword allows providing a custom "less
 than" function; use `rev=true` to reverse the sorting order.
 Items that compare equal receive the mean of the
 rankings they would have been assigned under ordinal ranking.
+Missing values are assigned rank `missing`.
 """
 tiedrank(x::AbstractArray; lt = isless, rev::Bool = false) =
     tiedrank!(Array{Float64}(uninitialized, size(x)), x, sortperm(x; lt = lt, rev = rev))
 
+for (f, f!, S) in zip([:ordinalrank, :competerank, :denserank, :tiedrank],
+                      [:ordinalrank!, :competerank!, :denserank!, :tiedrank!],
+                      [Int64, Int64, Int64, Float64])
+    @eval begin
+        function $f(x::AbstractArray{Union{T, Missing}}; lt = isless, rev::Bool = false) where {T}
+            xv = collect(skipmissing(x))
+            sp = sortperm(xv; lt = lt, rev = rev)
+            rks = missings($S, length(x))
+            $(f!)(view(rks, .!ismissing.(x)), xv, sp)
+            rks
+        end
+
+        $f(x::AbstractArray{Missing}; lt = isless, rev::Bool = false) = missings($S, length(x))
```

Comment:
Why is this needed? Because of ambiguities?

Reply:
My bad, it was actually not needed.